components/server/ee/src/user/user-service.ts

```diff
@@ -32,10 +34,35 @@ export class UserServiceEE extends UserService {
 
         // the self-hosted case
         if (!this.licenseEvaluator.isEnabled(Feature.FeatureSetTimeout, userCount)) {
-            return "30m";
+            return WORKSPACE_TIMEOUT_DEFAULT_SHORT;
         }
 
-        return "60m";
+        return WORKSPACE_TIMEOUT_DEFAULT_LONG;
+    }
+
+    public workspaceTimeoutToDuration(timeout: WorkspaceTimeoutDuration): string {
+        switch (timeout) {
+            case WORKSPACE_TIMEOUT_DEFAULT_SHORT:
+                return "30m";
+            case WORKSPACE_TIMEOUT_DEFAULT_LONG:
+                return this.config.workspaceDefaults.timeoutDefault || "60m";
+            case WORKSPACE_TIMEOUT_EXTENDED:
+            case WORKSPACE_TIMEOUT_EXTENDED_ALT:
+                return this.config.workspaceDefaults.timeoutExtended || "180m";
+        }
+    }
+
+    public durationToWorkspaceTimeout(duration: string): WorkspaceTimeoutDuration {
```

Comment:
I find the types a little confusing in these method signatures:
- `duration` is a `string`, and the returned `Timeout` is a `WorkspaceTimeoutDuration`? 🤔
- Also `workspaceTimeoutToDuration` takes a `WorkspaceTimeoutDuration` and turns it into a `string` (would expect the other way around)

I feel like `duration` should be of type `...Duration`, and `timeout` of type `...Timeout`. 💭

Maybe this could be a follow-up issue though. ⏩